plugins/sharding/logical_enumerator.rb

```diff
@@ -87,12 +122,20 @@ def each_shard_with_around(order)
           @shards << shard
           next if shards.size < 3
           yield(*shards)
-          shards.shift
+          shifted_shard = shards.shift
+          next unless unref_immediately
+          next if shifted_shard.nil?
```

Comment:
This is needless because `shards` must not be empty here.

Reply:
shards is initially `[nil]`, so the first shifted shard is `nil`.

https://github1s.com/groonga/groonga/blob/HEAD/plugins/sharding/logical_enumerator.rb#L70-L71

Reply:
Sorry.